node-fs-readfile/cat.js

```diff
@@ -0,0 +1,25 @@
+const fs = require('fs');
+
+let i = 2;
+
+function call(callback, file) {
+  fs.readFile(file, 'utf8', (err, data) => {
+    if (err) throw err;
+    console.log(data);
+  callback()
+  })
+}
+
+const readFile = (file) => {
+  fs.readFile(file, 'utf8', (err, data) => {
+    if (err) throw err;
+    console.log(data);
+  })
+}
+
+do {
+call(function() {
+  console.log(i);
+  }, process.argv[i])
+  i++;
+} while (i < process.argv.length)
```

Comment:
`readFile` is asynchronous, and each file read operation takes a different amount of time to complete. While this is guaranteed to _start_ the read operations in the correct order, there's no guarantee they'll _finish_ in the same order: https://lfz-lms.herokuapp.com/code-exercises/Learning-Fuze/curriculum/node-fs-readfile?v=4ffa7ec6e821c82e369116ef5b1a152a7dbf7bbc

---

![image](https://user-images.githubusercontent.com/16143771/127896014-1d256cfa-8137-4798-8397-e9d7f1c3b2f4.png)